fedimint-server/src/config/api.rs

```diff
@@ -317,17 +318,42 @@ impl ConfigGenApi {
         .map_err(|e| ApiError::server_error(format!("Unable to encrypt configs {e:?}")))
     }
 
-    /// Attempts to decrypt the config files from disk using the auth string and
-    /// notify the listeners
+    /// Attempts to decrypt the config files from disk using the auth string.
+    ///
+    /// Will force shut down the config gen API so the consensus API can start.
+    /// Removes the upgrade flag when called.
     pub async fn start_consensus(&self, auth: ApiAuth) -> ApiResult<()> {
         let cfg = read_server_config(&auth.0, self.data_dir.clone())
             .map_err(|e| ApiError::bad_request(format!("Unable to decrypt configs {e:?}")))?;
 
+        let mut tx = self.db.begin_transaction().await;
+        tx.remove_entry(&ConsensusUpgradeKey).await;
+        tx.commit_tx().await;
+
         self.config_generated_tx.send(cfg).await.expect("Can send");
 
         Ok(())
     }
 
+    /// Returns the server status
+    pub async fn server_status(&self) -> ServerStatus {
+        let has_upgrade_flag = { self.has_upgrade_flag().await };
```

Comment:
The braces seem a little bit unnecessary here?

Reply:
I'm used to doing that to keep the async away from the locks, but I think it is unnecessary...